resteasy-core/src/main/java/org/jboss/resteasy/core/ContextParameterInjector.java

```diff
@@ -38,13 +37,46 @@
 @SuppressWarnings("unchecked")
 public class ContextParameterInjector implements ValueInjector
 {
+   private static Constructor<?> constructor;
+
    private Class<?> rawType;
    private Class<?> proxy;
    private ResteasyProviderFactory factory;
    private Type genericType;
    private Annotation[] annotations;
    private volatile boolean outputStreamWasWritten = false;
 
+   static
+   {
+      constructor = AccessController.doPrivileged(new PrivilegedAction<Constructor<?>>()
+      {
+         @Override
+         public Constructor<?> run()
+         {
+            try
+            {
+               Class.forName("javax.servlet.http.HttpServletResponse", false, Thread.currentThread().getContextClassLoader());
+               Class<?> clazz = Class.forName("org.jboss.resteasy.core.ContextServletOutputStream");
+               return clazz.getDeclaredConstructor(ContextParameterInjector.class, OutputStream.class);
+            }
+            catch (Exception e)
+            {
+               return null;
+            }
+         }
+      });
+   }
+
+   public boolean isOutputStreamWasWritten()
+   {
+      return outputStreamWasWritten;
+   }
+
+   public void setOutputStreamWasWritten(boolean outputStreamWasWritten)
+   {
+      this.outputStreamWasWritten = outputStreamWasWritten;
+   }
```

Comment:
Minor and don't really need to be changed, but these could likely be package private.

Reply:
Believe it or not, I just came back to my desk to do that. :)